src/SSHDebugPS/Docker/DockerContainerInstance.cs

```diff
@@ -0,0 +1,124 @@
+    {
+        /// <summary>
+        /// Create a DockerContainerInstance from the results of docker ps in JSON format
+        /// </summary>
+        public static DockerContainerInstance Create(string json)
+        {
+            try
+            {
+                JObject obj = JObject.Parse(json);
+                var instance = obj.ToObject<DockerContainerInstance>();
+                if (instance != null)
+                    return instance;
+            }
+            catch (Exception e)
+            {
+                Debug.Fail(e.ToString());
+            }
+            return null;
+        }
+
+        private DockerContainerInstance() { }
+
+        #region JsonProperties
+
+        [JsonProperty("ID")]
+        public override string Id { get; set; }
+        public string ShortId { get => Id.Substring(0, 10); }
+
+        [JsonProperty("Names")]
+        public override string Name { get; set; }
+
+        [JsonProperty("Image")]
+        public string Image { get; private set; }
+
+        [JsonProperty("Ports")]
+        public string Ports { get; set; }
+
+        [JsonProperty("Command")]
+        public string Command { get; private set; }
+
+        [JsonProperty("Status")]
+        public string Status { get; private set; }
+
+        [JsonProperty("CreatedAt")]
+        public string Created { get; private set; }
+
+        #endregion
+
+        public override bool GetResult(out string selectedQualifier)
+        {
+            selectedQualifier = Name;
+            return true;
+        }
+
+        public override event PropertyChangedEventHandler PropertyChanged;
+
+        private void OnPropertyChanged(string name)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
+        }
+
+        public string FormattedListOfPorts
+        {
+            get
+            {
+                return string.IsNullOrWhiteSpace(Ports) ?
+                    UIResources.NoPortsText :
+                    Ports.Replace(", ", "\r\n");
+            }
+        }
+
+        private bool _isSelected;
+        public bool IsSelected
+        {
+            get
+            {
+                return _isSelected;
+            }
+            set
+            {
+                if (_isSelected != value)
+                {
+                    _isSelected = value;
+                    OnPropertyChanged(nameof(IsSelected));
+                }
+            }
+        }
+
+        protected override bool EqualsInternal(ContainerInstance instance)
+        {
+            if (instance is DockerContainerInstance other)
+            {
+                // the id can be a partial on a container
+                return String.Equals(Id, other.Id, StringComparison.OrdinalIgnoreCase) ? true :
+                    Id.StartsWith(other.Id, StringComparison.OrdinalIgnoreCase) ? true :
+                    other.Id.StartsWith(Id, StringComparison.OrdinalIgnoreCase) ? true : false;
```

Comment:
Writing this as a nested ternary seems kind of confusing - I had to read it a couple of times to make sure I understood what it was checking.  Could this be written as your 3 checks just ||'ed together?